### ISSUE TITLE
Two ideas

### ISSUE THREAD

**Issue by Moosems** [Dogeek/tkinter-pp#13]
First, can you make it possible to bind the notebook tab close to a function? Second, can you put this on PyPi?

**Comment by rdbende** [comment]
https://pypi.org/project/tkinterpp/

**Comment by Moosems** [comment]
Any way for the first?

**Comment by rdbende** [comment]
Looking at the source code, there's an event for that. You can bind your function to the `<<NotebookTabClosed>>` virtual event, that will be generated after the tab is closed.

**Comment by Moosems** [comment]
Hmm what data does the event send? Can I get the tab index from it?

**Comment by rdbende** [comment]
No. It doesn't send any data 😕
https://github.com/Dogeek/tkinter-pp/blob/a2a161c1993fc04c48840aca0b2886a50963f53f/tkinterpp/notebooks.py#L47

**Comment by Moosems** [comment]
How could I send data in a virtual event? I think I might make a pull request with some changes so that with both notebooks it sends data like the index of the closed tab or the index of the tab moved and which spot it went to.

NOTE:

I just found something that could work. If we have an attribute that has the index of the closed tab that changes before the event is sent then we can ask for the attributes value to get certain data. I also found [this](https://www.tcl.tk/man/tcl8.7/TkCmd/event.html#M5:~:text=not%20trigger%20anymore.-,event%20generate%20window%20event%20%3Foption%20value%20option%20value%20...%3F,require%20that%20the%20window%20has%20focus%20to%20receive%20the%20event%20properly.,-event%20info%20%3F)

**Comment by rdbende** [comment]
> I also found [this](https://www.tcl.tk/man/tcl8.7/TkCmd/event.html#M5)

Yes, there's a `data` field in Tk, but for some reason Tkinter doesn't support it. You can send data with a virtual event
```python
w.event_generate("<<Event>>", data=str(tab_index))
```
but the event object doesn't have that attribute
```python
>>> hasattr(event, "data")
False
```
If you really want to, you can some other field, but that's a bit hacky imo
```python
w.event_generate("<<Event>>", x=tab_index)

w.bind("<<Event>>", lambda event: print(f"Tab index: {event.x}"))
```

**Comment by Moosems** [comment]
Us there a way to call directly to tcl/tk for that?

**Comment by Moosems** [comment]
I found [this answer](https://stackoverflow.com/a/41912425/17053202) which could work
